podwatcher/extension.go

```diff
@@ -0,0 +1,112 @@
+package podwatcher
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"runtime"
+
+	eirinix "code.cloudfoundry.org/eirinix"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// https://github.com/cloudfoundry-incubator/eirini-staging/tree/master/image
+const (
+	defaultStagingExecutorEntrypoint   = "/packs/executor"
+	defaultStagingDownloaderEntrypoint = "/packs/downloader"
+	defaultStagingUploaderEntrypoint   = "/packs/uploader"
+	defaultRuntimeEntrypoint           = "/lifecycle/launch"
+	defaultFailGracePeriod             = "5"
+	defaultSuccessGracePeriod          = "5"
+)
+
+// GraceOptions lets customize the graceful periods and
+// the entrypoint of the images which are mutated to inject
+// the grace period logic
+type GraceOptions struct {
+	FailGracePeriod, SuccessGracePeriod string
+
+	StagingDownloaderEntrypoint string
+	StagingExecutorEntrypoint   string
+	StagingUploaderEntrypoint   string
+	RuntimeEntrypoint           string
+}
+
+// Extension changes pod definitions
+type Extension struct {
+	Logger  *zap.SugaredLogger
+	Options GraceOptions
+}
+
+// NewGracePeriodInjector returns the podwatcher extension which injects a grace Period on Eirini generated pods
+func NewGracePeriodInjector(opts *GraceOptions) *Extension {
+	if len(opts.StagingExecutorEntrypoint) == 0 {
```

Comment:
That's really about taste, but I think they are really both idiomatic, I found a lot of code even in standard packages using the `len(x) ==0` approach. That being said, I don't like to [fuel](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go) much those kinda of topics, as at the end is really just about what one likes most. 

It's just a habit of mine, but that to be honest I like more than a string comparison. To me looks more clear (heh, personal taste!) also because in other languages `len(x)` is less expensive than a string comparison when the compiler translate it behind the scenes, and that was also the case apparently with earlier Go version (apparently now it's  not the same anymore.. since 1.9?  )

Reply:
Yeah, I remember at some point comparing the assembly output for the two to check that they're actually identical.  But this is a taste thing, and I'm happy to keep `len(x) == 0`.